src/app/child/child.component.ts

```diff
@@ -19,15 +20,31 @@ export class ChildComponent {
       ngAfterViewInit,
       ngAfterViewChecked,
       ngOnDestroy,
-    } = getObservableLifecycle(this);
+    } =
+      // specifying the generics is only needed if you intend to
+      // use the `ngOnChanges` observable, this way you'll have
+      // typed input values instead of just a `SimpleChange`
+      getObservableLifecycle<ChildComponent, 'input1' | 'input2'>(this);
```

Comment:
what is the consequence of not defining the generic? `ngOnChanges` works fine but just has untyped object?